### PR TITLE
Adding a modal directive

### PR DESCRIPTION
**Pull request by pgiraud**
Please review.

### CODE REVIEW

File: examples/modal.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/** @type {!angular.Module} **/
+app.module = angular.module('app', ['ngeo']);
+
+
+
+/**
+ * @constructor
+ */
+app.MainController = function() {
+  this['modalShown'] = false;
```
Comment (elemoine) [inline examples/modal.js]: We now use `@export`.

```js
    /**
     * @type {boolean}
     * @export
     */
    this.modalShown = false;
```

The `@export` annotation prevents renaming.

File: src/directives/modal.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @fileoverview Provides the "ngeoModal" directive. This directive allows
+ * displaying a modal window. The directive takes care or the show/hide
+ * actions.
+ *
+ *
+ * Example:
+ *
+ * <modal shown="modalShown">
```
Comment (elemoine) [inline src/directives/modal.js]: `<modal>` -> `<ngeo-modal>`

Comment (elemoine) [inline src/directives/modal.js]: Even:

`<modal shown="modalShown">`  -> `<ngeo-modal ngeo-modal-shown="modalShown">`. This is to match our conventions.

File: src/directives/modal.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    template: '<div class="modal fade" tabindex="-1" role="dialog">' +
+        '<div class="modal-dialog">' +
+        '<div class="modal-content">' +
+        '<ng-transclude></ng-transclude>' +
+        '</div>' +
+        '</div>' +
+        '</div>',
+    restrict: 'E',
+    transclude: true,
+    replace: true,
```
Comment (elemoine) [inline src/directives/modal.js]: `replace` is deprecated. We don't use it in ngeo.

File: src/directives/modal.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ */
+
+goog.provide('ngeo.modalDirective');
+
+goog.require('ngeo');
+
+
+/**
+ * @param {angular.$parse} $parse Angular parse service.
+ * @return {angular.Directive} The directive specs.
```
Comment (elemoine) [inline src/directives/modal.js]: You need to use `@ngInject` or it won't work when compiled.

File: src/directives/modal.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        '<div class="modal-dialog">' +
+        '<div class="modal-content">' +
+        '<ng-transclude></ng-transclude>' +
+        '</div>' +
+        '</div>' +
+        '</div>',
+    restrict: 'E',
+    transclude: true,
+    replace: true,
+    link: function(scope, element, attrs) {
```
Comment (elemoine) [inline src/directives/modal.js]: I think it's a good idea to type the function. See https://github.com/camptocamp/ngeo/blob/master/src/directives/btngroup.js#L107-L113. In this way, `scope`, `element` and `attrs` are typed, and you're then guaranteed to 
correctly use these variables.

File: src/directives/modal.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @fileoverview Provides the "ngeoModal" directive. This directive allows
+ * displaying a modal window. The directive takes care or the show/hide
+ * actions.
```
Comment (elemoine) [inline src/directives/modal.js]: Is the directive bootstrap-specific? If so it should be documented. Also, does the directive requires bootstrap.js and jQuery? If so it should be documented as well.

File: src/directives/modal.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+ */
+ngeo.modalDirective = function($parse) {
+  return {
+    template: '<div class="modal fade" tabindex="-1" role="dialog">' +
+        '<div class="modal-dialog">' +
+        '<div class="modal-content">' +
+        '<ng-transclude></ng-transclude>' +
+        '</div>' +
+        '</div>' +
+        '</div>',
```
Comment (elemoine) [inline src/directives/modal.js]: I'd recommend adding a partial.

**Comment by elemoine** [comment]
I added a couple comments, but it looks good to me. I think you can merge this when the comments are addressed. Thanks!

File: src/directives/modal.js (hunk 6 above)
Comment (pgiraud) [inline src/directives/modal.js]: There's a note for this a few lines below.

File: src/directives/modal.js (hunk 7 above)
Comment (pgiraud) [inline src/directives/modal.js]: Using a partial was a bit painful in this simple case.

**Pull request merged**